payment-method-basic-card/billing-address-is-null-manual.https.html

```diff
@@ -0,0 +1,150 @@
+  const details = {
+    total: {
+      label: "label",
+      amount: { currency: "USD", value: "5.00" },
+    },
+  };
+
+  // Smoke tests
+  test(() => {
+    assert_true(
+      "onpaymentmethodchange" in PaymentRequest.prototype,
+      "The paymentmethodchange event handler is not supported on PaymentRequest"
+    );
+    assert_true(
+      "PaymentMethodChangeEvent" in window,
+      "The PaymentMethodChangeEvent is not supported"
+    );
+  }, "PaymentMethodChangeEvent support");
+
+  function dontRequestBillingAddress(options) {
+    promise_test(async t => {
+      const request = new PaymentRequest([basicCard], details, {
+        requestBillingAddress: false,
+      });
+      const showPromise = request.show();
+
+      // Let's check the method data from PaymentMethodChangeEvent.
+      const event = await new Promise(resolve =>
+        request.addEventListener("paymentmethodchange", resolve)
+      );
+      assert_true(
+        event instanceof PaymentMethodChangeEvent,
+        "Expected instance of PaymentMethodChangeEvent"
+      );
+      assert_equals(
+        methodDetails.billingAddress,
+        null,
+        "Expected methodDetails.billingAddress to be null"
+      );
+
+      // Let's check the billingAddress in the response
+      const response = await showPromise;
+      const {
+        details: { billingAddress: responseBillingAddress },
+      } = response;
+
+      assert_equals(
+        responseBillingAddress,
+        null,
+        "Expected PaymentResponse.data.billingAddress to be null"
+      );
+
+      // And we are done
+      await response.complete("success");
+    });
+  }
+
+  function requestBillingAddress() {
+    promise_test(async t => {
+      const request = new PaymentRequest([basicCard], details, {
+        requestBillingAddress: true,
+      });
+      const showPromise = request.show();
+
+      // Let's check the methodDetails from event.
+      const event = await new Promise(resolve =>
+        request.addEventListener("paymentmethodchange", resolve)
+      );
+      assert_true(
+        event instanceof PaymentMethodChangeEvent,
+        "Expected instance of PaymentMethodChangeEvent"
+      );
+      const { billingAddress: eventBillingAddress } = event.methodDetails;
+      checkRedactList(eventBillingAddress);
+
+      // Let's check the billingAddress in the response.
+      const response = await showPromise;
+      const {
+        details: { billingAddress: responseBillingAddress },
+      } = await showPromise;
+      checkRedactList(responseBillingAddress);
+
+      // And we are done.
+      await response.complete("success");
+    });
+  }
+
+  function checkRedaction(billingAddress) {
+    assert_true(
+      billingAddress instanceof PaymentAddress,
+      "Expected instance of PaymentAddress"
+    );
+    for (const item of ["organization", "phone", "recipient"]) {
+      assert_equals(
+        billingAddress[item],
+        null,
```

Comment:
Is `undefined` an accepted outcome of redaction?

Reply:
hmmm... going point... this is wrong, but for a different reason: In the PR spec, these all come back as empty strings. However, that's not very clear. What we should do is make `AddressInit`'s members all default to the empty string.

Reply:
Sent pull request https://github.com/w3c/payment-request/pull/852